src/EditorFeatures/CSharpTest/ConvertToRecord/ConvertToRecordTests.cs

```diff
@@ -0,0 +1,1886 @@
+        {
+            var initialMarkup = @"
+namespace N
+{
+    public struct [|C|]
+    {
+        public int P { get; init; }
+    }
+}
+";
+            await TestNoRefactoringAsync(initialMarkup).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestMoveSimpleProperty()
+        {
+            var initialMarkup = @"
+namespace N
+{
+    public class [|C|]
+    {
+        public int P { get; init; }
+    }
+}
+";
+            var changedMarkup = @"
+namespace N
+{
+    public record C(int P);
+}
+";
+            await TestRefactoringAsync(initialMarkup, changedMarkup).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestReadonlyProperty()
+        {
+            var initialMarkup = @"
+namespace N
+{
+    public class [|C|]
+    {
+        public int P { get; }
+    }
+}
+";
+            var fixedMarkup = @"
+namespace N
+{
+    public record [|C|](int P);
+}
+";
+            await TestRefactoringAsync(initialMarkup, fixedMarkup).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestInitPropertyOnReadonlyStruct()
+        {
+            var initialMarkup = @"
+namespace N
+{
+    public readonly struct [|C|]
+    {
+        public int P { get; init; }
+    }
+}
+";
+            var fixedMarkup = @"
+namespace N
+{
+    public readonly record struct [|C|](int P);
+}
+";
+            await TestRefactoringAsync(initialMarkup, fixedMarkup).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestMovePropertySimpleInheritance()
+        {
+            var initialMarkup = @"
+namespace N
+{
+    public class B
+    {
+    }
+
+    public class [|C|] : B
+    {
+        public int P { get; init; }
+    }
+}
+";
+            // three of the same error on C because the generated
+            // EqualityConstract, Equals, and PrintMembers are all declared override
+            // and there's nothing to override.
+            // The other errors are copy constructor expected in B, and the
+            // "records can't inherit from class" on B as well
+            var changedMarkup = @"
+namespace N
+{
+    public class B
+    {
+    }
+
+    public record {|CS0115:{|CS0115:{|CS0115:{|CS8867:C|}|}|}|}(int P) : {|CS8864:B|};
+}
+";
+            await TestRefactoringAsync(initialMarkup, changedMarkup).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestMovePropertySimpleInterfaceInheritance()
+        {
+            var initialMarkup = @"
+namespace N
+{
+    public interface IInterface
+    {
+        public int Foo();
+    }
+
+    public class [|C|] : IInterface
+    {
+        public int P { get; init; }
+
+        public int Foo()
+        {
+            return P;
+        }
+    }
+}
+";
+            var changedMarkup = @"
+namespace N
+{
+    public interface IInterface
+    {
+        public int Foo();
+    }
+
+    public record C(int P) : IInterface
+    {
+
```

Comment:
the property retains the leading newline trivia from the original definition. Since the original definition had a blank line between the property and method, the method keeps the newline. It might be useful to elasticize the whitespace or format somehow, but I don't want to remove whitespace that the user may have intentionally added.

Reply:
I think removing whitespace trivia from first member and keeping all other trivia might be a good option.